src/features/goblins/rocket/Rocket.tsx

```diff
@@ -0,0 +1,387 @@
+import { melonDuskAudio, rocketLaunchAudio } from "lib/utils/sfx";
+import momNpc from "assets/mom/mom_npc.gif";
+import scaffoldingLeft from "assets/mom/scaffolding_left.png";
+import scaffoldingRight from "assets/mom/scaffolding_right.png";
+import support from "assets/mom/launch-pad-material-2.png";
+import platform from "assets/mom/launch-pad-material-3.png";
+import woodPile from "assets/mom/launch-pad-material-4.png";
+import goblinHammering from "assets/mom/goblin_mechanic_1.gif";
+import goblinWelding from "assets/mom/goblin_mechanic_2.gif";
+import goblinForeman from "assets/mom/goblin_mechanic_3.gif";
+import metalSheetsPileFew from "assets/mom/metal-sheets-pile-few.png";
+import metalSheetsPileMany from "assets/mom/metal-sheets-pile-many.png";
+
+import { GRID_WIDTH_PX } from "features/game/lib/constants";
+import { Context } from "features/game/GoblinProvider";
+
+const ROCKET_LAUNCH_TO_DIALOG_TIMEOUT = 4000;
+const MELON_DUSK_SEEN = "isMelonDuskSeen";
+
+export const Rocket: React.FC = () => {
+  const { goblinService } = useContext(Context);
+  const [
+    {
+      context: { state },
+    },
+  ] = useActor(goblinService);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [isItemsOpen, setIsItemsOpen] = useState(false);
+  const [isRocketLaunching, setIsRocketLaunching] = useState(false);
+  const [isRocketLaunchComplete, setIsRocketLaunchComplete] = useState(false);
+  const [hasCompletedMission, setHasCompletedMission] = useState(false);
+  const [isMintingObservatory, setIsMintingObservatory] = useState(false);
+  const [observatoryMintError, setObservatoryMintError] =
+    useState<Error | null>(null);
+
+  const isRocketFixed = (state.inventory["Engine Core"]?.toNumber() || 0) > 0;
+  const hasCompletedQuest = (state.inventory.Observatory?.toNumber() || 0) > 0;
+
+  // Check if player has already completed mission
+  useEffect(() => {
+    (async () => {
+      const isComplete = await metamask
+        .getMillionOnMars()
+        .hasCompletedMission();
+
+      setHasCompletedMission(isComplete);
+      // If player has already completed mission, then the launch is also complete
+      setIsRocketLaunchComplete(isComplete);
+    })();
+  }, []);
+
+  useEffect(() => {
+    if (!isRocketLaunching) {
+      return;
+    }
+    melonDuskAudio.stop();
+    rocketLaunchAudio.play();
+    setTimeout(() => {
+      setIsRocketLaunching(false);
+      setIsRocketLaunchComplete(true);
+      setIsDialogOpen(true);
+      melonDuskAudio.play();
+    }, ROCKET_LAUNCH_TO_DIALOG_TIMEOUT);
+  }, [isRocketLaunching]);
+
+  const handleLaunchRocket = () => {
+    setIsDialogOpen(false);
+    if (isRocketLaunching) {
+      // Don't launch again if it's already launching
+      return;
+    }
+    setIsRocketLaunching(true);
+  };
+
+  const handleOpenDialog = () => {
+    setIsDialogOpen(true);
+    if (
+      isMelonDuskSeen &&
+      !isRocketFixed &&
+      !hasCompletedMission &&
+      !hasCompletedQuest
```

Comment:
This is a good example where having too many states in a component becomes hard to manage.

If a new state was to be added, this component would get even more complex and increase the chance of a regression